dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -613,6 +614,65 @@ Spdp::get_ice_endpoint_if_added()
   return tport_->ice_endpoint_added_ ? tport_->get_ice_endpoint() : 0;
 }
 
+bool is_ip_equal(const ACE_INET_Addr& a, const DCPS::Locator_t& locator)
+{
+  const ACE_UINT32 a_ip = ntohl(a.get_ip_address());
```

Comment:
This approach won't work for when `a` is a real IPv6 address, since IPv6 addresses are 128 bits (not 32).

What you will probably want to do instead is use ACE_INET_Addr::get_addr() and cast the result (void *) to a  sockaddr* check for family / address pointer that way. You will specifically need to handle the case when one address is IPv4 and the other is IPv4-mapped-IPv6 (different kinds) but the last four bytes of the IPv6 address match the IPv4 address.

Reply:
ACE_INET_Addr::get_ip_address() handles the type issue and byte order, and returns a 4-byte ip address:
 
// Return the 4-byte IP address, converting it into host byte order.
ACE_UINT32
ACE_INET_Addr::get_ip_address (void) const
{
  ACE_TRACE ("ACE_INET_Addr::get_ip_address");
#if defined (ACE_HAS_IPV6)
  if (this->get_type () == AF_INET6)
    {
      if (IN6_IS_ADDR_V4MAPPED (&this->inet_addr_.in6_.sin6_addr) ||
          IN6_IS_ADDR_V4COMPAT (&this->inet_addr_.in6_.sin6_addr)    )
        {
          ACE_UINT32 addr;
          // Return the last 32 bits of the address
          char *thisaddrptr = (char*)this->ip_addr_pointer ();
          thisaddrptr += 128/8 - 32/8;
          ACE_OS::memcpy (&addr, thisaddrptr, sizeof (addr));
          return ACE_NTOHL (addr);
        }

      ACELIB_ERROR ((LM_ERROR,
                  ACE_TEXT ("ACE_INET_Addr::get_ip_address: address is a IPv6 address not IPv4\n")));
      errno = EAFNOSUPPORT;
      return 0;
    }
#endif /* ACE_HAS_IPV6 */
  return ACE_NTOHL (ACE_UINT32 (this->inet_addr_.in4_.sin_addr.s_addr));
}

Reply:
removed call to get_ip_address